Assets/MixedRealityToolkit-SDK/Features/UX/Interactable/Scripts/InspectorFields/InspectorField.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX
+{
+    /// <summary>
+    /// A set of Inspector fields for setting up properties in a
+    /// component to automatically draw in a custom inspector
+    /// </summary>
+    public class InspectorField : Attribute
+    {
+        public enum FieldTypes { Float, Int, String, Bool, Color, DropdownInt, DropdownString, GameObject, ScriptableObject, Object, Material, Texture, Vector2, Vector3, Vector4, Curve, Quaternion, AudioClip, Event }
+
+        public FieldTypes Type { get; set; }
+        public string Label { get; set; }
+        public string Tooltip { get; set; }
+        public string[] Options { get; set; }
+        public UnityEngine.Object Value { get; set; }
+
+        public static PropertySetting FieldToProperty(InspectorField attributes, object fieldValue, string fieldName)
+        {
+            PropertySetting setting = new PropertySetting();
+            setting.Type = attributes.Type;
+            setting.Tooltip = attributes.Tooltip;
+            setting.Label = attributes.Label;
+            setting.Options = attributes.Options;
+            setting.Name = fieldName;
+
+            UpdatePropertySetting(setting, fieldValue);
+
+            return setting;
+        }
+        
+        public static PropertySetting UpdatePropertySetting(PropertySetting setting, object update)
+        {
+            switch (setting.Type)
```

Comment:
I don't know for sure that this exists, but is there not a way to do something like this:
(setting.Type)update;

Why do we need to keep the setting value in these type specific fields? Only one of them can be set at once, right?

Reply:
You are correct, I built it out that way to figure out how many different properties I would need for the enum and if there were any special cases which would require a switch statement. The event was the only thing I found so far that did not work as planned. I thought about condensing it during my clean up last week and feel better about it now after getting your input.